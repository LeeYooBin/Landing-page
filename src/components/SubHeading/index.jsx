import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../styles/theme';

const Content = styled.h2`
  font: ${theme.fonts.heading4};
  color: ${theme.colors.sun};
  text-align: center;
  line-height: 3.2rem;
  letter-spacing: 5px;
  text-transform: uppercase;

  @media screen and (min-width: 920px) {
    font: ${theme.fonts.heading3};
    text-align: left;
  }
`;

const SubHeading = ({ text = '' }) => <Content>{text}</Content>;

SubHeading.propTypes = {
  text: PropTypes.string,
};

export default SubHeading;
