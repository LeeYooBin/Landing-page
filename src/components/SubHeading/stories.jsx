import SubHeading from '.';

export default {
  title: 'SubHeading',
  component: SubHeading,
};

export const Template = (args) => <SubHeading {...args} />;
