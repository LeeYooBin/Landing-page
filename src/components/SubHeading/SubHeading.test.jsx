import { describe, it, expect } from '@jest/globals';
import { screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import SubHeading from '.';
import renderTheme from '../../styles/render-theme';

describe('Heading component', () => {
  it('renders without errors', () => {
    renderTheme(<SubHeading />);
  });

  it('renders the correct text when isMain is false', () => {
    const { getByText } = renderTheme(<SubHeading text="Subheading" />);
    expect(getByText('Subheading')).toBeInTheDocument();
  });

  it('renders the default text when no text prop is provided', () => {
    expect(screen.queryByText('')).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const { container } = renderTheme(<SubHeading text="Text" />);
    expect(container).toMatchSnapshot();
  });

  it('validates the prop types correctly', () => {
    const consoleError = console.error;
    // eslint-disable-next-line no-undef
    console.error = jest.fn();

    renderTheme(<SubHeading text={123} />);
    expect(console.error).toHaveBeenCalled();
    expect(console.error).toHaveBeenCalledTimes(1);

    console.error.mockRestore();
    console.error = consoleError;
  });
});
