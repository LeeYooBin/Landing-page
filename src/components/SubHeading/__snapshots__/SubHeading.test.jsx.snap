// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component Should match snapshot 1`] = `
.c0 {
  font: 400 1.4rem/1.8rem "Heebo",sans-serif;
  color: #F5D15F;
  text-align: center;
  line-height: 3.2rem;
  letter-spacing: 5px;
  text-transform: uppercase;
}

@media screen and (min-width: 920px) {
  .c0 {
    font: 400 2rem/2.4rem "Heebo",sans-serif;
    text-align: left;
  }
}

<div>
  <h2
    class="c0"
  >
    Text
  </h2>
</div>
`;
